protocols/dfns-cggmp21/src/lib.rs

```diff
@@ -17,143 +19,102 @@ pub mod error;
 pub mod protocols;
 pub mod util;
 
-#[protocol]
-pub struct DfnsCGGMP21KeygenProtocolConfig<
-    N: Network,
-    B: Block,
-    BE: Backend<B>,
-    KBE: KeystoreBackend,
-    C: ClientWithApi<B, BE>,
-> where
-    <C as ProvideRuntimeApi<B>>::Api: JobsApi<B, AccountId>,
-{
-    pub account_id: AccountId,
-    pub network: N,
-    pub keystore_backend: ECDSAKeyStore<KBE>,
-    pub client: C,
-    pub logger: DebugLogger,
-    pub pallet_tx: Arc<dyn PalletSubmitter>,
-    pub _pd: std::marker::PhantomData<(B, BE)>,
-}
+/// A Helper macro to declare a protocol, used
+/// to avoid code duplication.
+macro_rules! decl_porto {
```

Comment:
Should this be abstracted into common macros? Usable in other protocols? CC @tbraun96